src/components/certificates/CertificateForm.tsx

```diff
@@ -86,7 +104,7 @@ const NewCertificateForm: React.FC = () => {
 
   return (
     <div className="form-container">
-      <h2>New Certificate</h2>
+      <h2>{certificate ? 'Edit Certificate' : 'New Certificate'}</h2>
```

Comment:
you should use some other identifier weather it is edit or create, not by just checking the whole object if it's present. Especially if you do changes from my previous comments.